core/src/main/java/org/opensearch/sql/data/utils/MaxTypeComparator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.data.utils;
+
+import java.util.Comparator;
+
+/** Comparator for MAX operations where strings have higher precedence than numbers. */
+public class MaxTypeComparator implements Comparator<Object> {
```

Comment:
Is `MinTypeComparator` simply the inverse of `MaxTypeComparator`? I'm thinking of only one comparator and use it in Min/MaxFunction below by Java Stream's min/max(comparator) API.

Also this may worth adding dedicated UT to show its behavior for num-num, num-string, string-string etc.